core/src/main/java/org/apache/brooklyn/core/objs/BasicSpecParameter.java

```diff
@@ -384,5 +387,73 @@ public static WeightedParameter getFieldConfig(ConfigKey<?> config, Field config
         }
     }
 
+    /**
+     * Returns true if the {@link SpecParameter parameter name} is the same as
+     * that on the specified paramater.
+     */
+    protected static Predicate<SpecParameter<?>> sameName(final SpecParameter<?> param) {
+        return new Predicate<SpecParameter<?>>() {
+            @Override
+            public boolean apply(SpecParameter<?> input) {
+                return input.getConfigKey().getName().equals(param.getConfigKey().getName());
+            }
+        };
+    }
+
+    /**
+     * Returns true if the {@link SpecParameter#getLabel() label} is the same as
+     * the specified string.
+     */
+    public static Predicate<SpecParameter<?>> labelEqualTo(final String label) {
+        return new Predicate<SpecParameter<?>>() {
```

Comment:
I'd be tempted to move the public methods to a separate `SpecParameterPredicates` class.